### PR TITLE
Initial meal calender UI done. 

### PR DESCRIPTION
**Pull request by ZSank**
# Pull Request

## Description

Please include a summary of the changes you have made. Highlight the key aspects of your project.

1. Mostly made changes in mealmate/view/utils/custom_views/MealCalender.kt
2. This file contains the composable, and also the logic, which handlers the dates. 

## Checklist

- [ ] My project includes a `README.md` file with a brief overview of the project, how to set it up, and any relevant information.
- [x] I have added a screenshot or screen recording showing the functionality of the app.
- [x] I have tested the project and confirmed it works as expected.

## Screenshots

If applicable, please add a screenshot or a screen recording to demonstrate the functionality of your app.

## Project README

Please ensure that your project's `README.md` is detailed and includes:

- A brief introduction to the project.
- Installation steps.
- Key features.
- Any additional instructions or information.

## Additional Notes

Is there anything else the maintainers should know about your contribution?

---

Thank you for your contribution!

### CODE REVIEW

**Comment by ZSank** [comment]
I know, I must move the CalendarUiModel data class and the CalendarDataSource class to some file. But I am not able to decide, where should I keep them. So, for now, I kept them in the MealCalender file itself.

File: meal_mate/app/src/main/java/com/android/engineer/mealmate/view/features/home/HomeScreen.kt [hunk 1]
```diff
@@ -80,6 +81,7 @@ fun ShowTopView(userName: String) {
 }
 
 @Composable
+@Preview(showBackground = true)
```
Comment (Siduchels) [inline meal_mate/app/src/main/java/com/android/engineer/mealmate/view/features/home/HomeScreen.kt:84]: Remove the preview annotation.

File: meal_mate/app/src/main/java/com/android/engineer/mealmate/view/features/meal_plan/MealPlanScreen.kt [hunk 2]
```diff
@@ -21,9 +22,10 @@ fun MealPlanScreen(title: String, navHostController: NavHostController) {
            .fillMaxSize()
            .background(OrangeOnPrimary),
        horizontalAlignment = Alignment.CenterHorizontally,
-       verticalArrangement = Arrangement.Center
    ) {
-        Text(title)
+	   MealCalender(
+		   modifier = Modifier.padding(10.dp).padding(top = 20.dp)
```
Comment (Siduchels) [inline meal_mate/app/src/main/java/com/android/engineer/mealmate/view/features/meal_plan/MealPlanScreen.kt:27]: remove one padding() from here.

File: meal_mate/app/src/main/java/com/android/engineer/mealmate/view/utils/custom_views/MealCalender.kt [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+		)
+	}
+}
+
+data class CalendarUiModel(
+	val selectedDate: Date,
+	val visibleDates: List<Date>
+) {
+	
+	val startDate: Date = visibleDates.first()
```
Comment (Siduchels) [inline meal_mate/app/src/main/java/com/android/engineer/mealmate/view/utils/custom_views/MealCalender.kt:219]: Move CalendarUiModel class to meal_plan/model folder.

File: meal_mate/app/src/main/java/com/android/engineer/mealmate/view/utils/custom_views/MealCalender.kt [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+		val day: String =
+			date.format(DateTimeFormatter.ofPattern("E"))
+	}
+}
+
+class CalendarDataSource {
+	val today: LocalDate
+		get() {
+			return LocalDate.now()
+		}
```
Comment (Siduchels) [inline meal_mate/app/src/main/java/com/android/engineer/mealmate/view/utils/custom_views/MealCalender.kt:236]: Move CalendarDataSource class to meal_plan folder.

File: meal_mate/app/src/main/java/com/android/engineer/mealmate/view/utils/custom_views/MealCalender.kt [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+	val dataSource = CalendarDataSource()
+	var data by remember { mutableStateOf(dataSource.getData(lastSelectedDate = dataSource.today)) }
+	Column(modifier = modifier.fillMaxWidth()) {
+		Spacer(modifier = Modifier.height(10.dp))
+		CalenderHeader(
+			data = data,
+			onPrevClickListener = { startDate ->
+				val finalStartDate = startDate.minusDays(1)
+				data = dataSource.getData(
+					startDate = finalStartDate,
```
Comment (Siduchels) [inline meal_mate/app/src/main/java/com/android/engineer/mealmate/view/utils/custom_views/MealCalender.kt:63]: Movie the all the business logic to ViewModel.

**Review by Siduchels — CHANGES_REQUESTED** [review]

**Pull request closed without merging**